bundles/org.openhab.binding.sonnen/src/main/java/org/openhab/binding/sonnen/internal/SonnenHandler.java

```diff
@@ -0,0 +1,276 @@
+            SonnenJsonDataDTO data = serviceCommunication.getBatteryData();
+            if (data != null) {
+                switch (channelId) {
+                    case CHANNELBATTERYDISCHARGING:
+                        update(OnOffType.from(data.isBatteryDischarging()), channelId);
+                        break;
+                    case CHANNELBATTERYCHARGING:
+                        update(OnOffType.from(data.isBatteryCharging()), channelId);
+                        break;
+                    case CHANNELCONSUMPTION:
+                        state = new QuantityType<Power>(data.getConsumptionHouse(), Units.WATT);
+                        update(state, channelId);
+                        break;
+                    case CHANNELGRIDFEEDIN:
+                        state = new QuantityType<Power>(data.getGridFeedIn(), Units.WATT);
+                        update(state, channelId);
+                        break;
+                    case CHANNELSOLARPRODUCTION:
+                        state = new QuantityType<Power>(data.getSolarProduction(), Units.WATT);
+                        update(state, channelId);
+                        break;
+                    case CHANNELBATTERYLEVEL:
+                        state = new QuantityType<Dimensionless>(data.getBatteryChargingLevel(), Units.PERCENT);
+                        update(state, channelId);
+                        break;
+                    case CHANNELFLOWCONSUMPTIONBATTERY:
+                        update(OnOffType.from(data.isFlowConsumptionBattery()), channelId);
+                        break;
+                    case CHANNELFLOWCONSUMPTIONGRID:
+                        update(OnOffType.from(data.isFlowConsumptionGrid()), channelId);
+                        break;
+                    case CHANNELFLOWCONSUMPTIONPRODUCTION:
+                        update(OnOffType.from(data.isFlowConsumptionProduction()), channelId);
+                        break;
+                    case CHANNELFLOWGRIDBATTERY:
+                        update(OnOffType.from(data.isFlowGridBattery()), channelId);
+                        break;
+                    case CHANNELFLOWPRODUCTIONBATTERY:
+                        update(OnOffType.from(data.isFlowProductionBattery()), channelId);
+                        break;
+                    case CHANNELFLOWPRODUCTIONGRID:
+                        update(OnOffType.from(data.isFlowProductionGrid()), channelId);
+                        break;
+                }
+            }
+        }
+    }
+
+    /**
+     * Updates the State of the given channel
+     *
+     * @param state
+     * @param channelId
+     */
+    private void update(@Nullable State state, String channelId) {
+        logger.debug("Update channel {} with state {}", channelId, (state == null) ? "null" : state.toString());
+
+        if (state != null) {
+            updateState(channelId, state);
+
+        } else {
+            updateState(channelId, UnDefType.NULL);
```

Comment:
NULL is not a state to be set. Rather use UNDEF.

Reply:
ok